src/vmm/src/device_manager/mmio.rs

```diff
@@ -234,6 +229,17 @@ impl MMIODeviceManager {
         self.register_mmio_device(identifier, slot, Arc::new(Mutex::new(device)))
     }
 
+    /// Create and register a boot timer device.
+    pub fn register_new_mmio_boot_timer(&mut self, request_ts: TimestampUs) -> Result<()> {
+        // Create and attach a new boot timer device.
+        let slot = self.allocate_new_slot()?;
```

Comment:
You can modify `allocate_new_slot()` to take `irq_count` as a param and allocate the right number of IRQs - in this case `0`.

Reply:
Done,  although I have some reservations about the implementation.